components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -177,6 +187,35 @@ internal abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
         }
     }
 
+    /**
+     * Return a set of the all the labels in use for a given labeled metric.
+     * This includes labels in all of the pings specified by the metric, and
+     * for all lifetimes.
+     *
+     * @param metricData The labeled metric to find labels for.
+     * @return The set of labels currently in use.
+     */
+    @Suppress("NestedBlockDepth")
+    override fun getLabelsForMetric(metricData: CommonMetricData): Set<String> {
```

Comment:
This shouldn't really live in the GenericStorageEngine, IMHO. I'd suggest to return the PR to the previous state (with the knonw issue about persisted labelled data), merge that, and approach this as follow-up work as this might require some discussion.

Reply:
Ok.  I've filed https://bugzilla.mozilla.org/show_bug.cgi?id=1527312 to track this.